flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/HiveCatalog.java

```diff
@@ -224,7 +225,11 @@ private static HiveConf createHiveConf(
         } else {
             hadoopConf = getHadoopConfiguration(hadoopConfDir);
         }
-        if (hadoopConf == null) {
+        if (hadoopConf == null && !isNullOrWhitespaceOnly(hadoopConfDir)) {
```

Comment:
These can be merged into the above if/else branches. No need to check `isNullOrWhitespaceOnly` again.